xml/hwbp_registers.xml

```diff
@@ -262,8 +260,19 @@
     </register>
 
     <register name="Machine Context" short="mcontext" address="0x7a8">
-        This optional register is an alias for \RcsrHcontext and is only
-        accessible in M-mode and Debug mode.
+        This register must be implemented if \RcsrHcontext is implemented, and
+        is optional otherwise.  It is only accessible in M-mode and Debug mode.
+
+        This register is an alias of the \RcsrHcontext register, and provides
+        access to the \FcsrHcontextHcontext field even if the \RcsrHcontext
+        register is not implemented.
```

Comment:
This seems confusing to me.  It's an alias of a register that's not implemented.

I was going to suggest that we say: If hcontext is implemented then this is an alias of that.  If hcontext is not implemented then this is its own register.

Then it occurred to me that there's no definition of mcontext fields in the latter scenario.  Maybe we should reverse things and define the fields here.  Then mcontext doesn't need to say anything about hcontext except that it must exist if hcontext exists (which it says in the previous paragraph).  And hcontext simply says that it's an optional alias of mcontext and which modes it's accessible in.

The only complication is that the field name within this register is named hcontext.  But maybe that's OK if we explain that one primary purpose of mcontext is for hypervisors running in M mode.

Reply:
I'll flip it like you suggest, and see if we like that better.